lib/iris/analysis/__init__.py

```diff
@@ -582,7 +582,11 @@ def aggregate(self, data, axis, **kwargs):
         mdtol = kwargs.pop("mdtol", None)
 
         result = self.call_func(data, axis=axis, **kwargs)
-        if mdtol is not None and ma.isMaskedArray(data):
+        if (
+            mdtol is not None
+            and ma.is_masked(data)
```

Comment:
Changing from `isMaskedArray` to `is_masked` isn’t part of the fix, but seems like the better check: if there is an all-False mask, then the loop won’t have any effect, so more efficient to just skip it.